deptofdefense/solo
# Feat 76 Configure development environment / proxy API calls to backend

Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist
I have…
- [x] review and update SSAG documentation if needed.


## Summary of Changes
This pull request…
- Add docker-compose configuration for development environment with hot reloading
- Dynamically generate NGINX configuration based on environment variables
- Create/test RootAPI connection class for frontend API calls
- Add cors header support to django for API calls from different domain
- Configure allowed cors hosts dynamically via environment
- Add a "API Test" button that makes an API call for validating that the front and backend can communicate
- Update CONTRIBUTING.md with instructions for starting development environment

## Testing
To verify the changes proposed in this pull request…
1. Start local development environment via the following instructions taken from the CONTRIBUTING.md instructions:

- A docker-compose.yml file is included in the repository that quickly and easily allows the execution of the entire system for rapid local development. (all commands are specified from the `deploy/compose` directory):
  1. Copy the example environment file to a new file called `.env`
      -  `cp example.env .env`
  2. Edit the `.env` file and create a username and password for the database connection, and set your secret key to be a random value.
      - Note these values will only apply to your local environment
  3. Execute `docker-compose up --build`
  4. After the build process, navigate to `http://localhost:3000` in a browswer
      - Note that hot reloading is enabled for changes to front and backend source files

## Screenshots
<img width="925" alt="Screen Shot 2020-02-04 at 4 22 24 PM" src="https://user-images.githubusercontent.com/59582489/73788198-9af3dd00-476a-11ea-85e1-fa0332cfabed.png">
